include/sys/ddt_impl.h

```diff
@@ -41,6 +41,17 @@ extern "C" {
 #define	DDT_DIR_VERSION		"version"
 #define	DDT_DIR_FLAGS		"flags"
 
+/* Fill a lightweight entry from a live entry. */
+#define	DDT_ENTRY_TO_LIGHTWEIGHT(ddt, dde, ddlwe) do {		\
+	memset((ddlwe), 0, sizeof (*ddlwe));			\
```

Comment:
Technically this memset() isn't needed.  However, I too typically prefer the "better safe than sorry" memsets like this just in case someone modifies `ddt_lightweight_entry_t` without changing the macro.  So keep it in, but we should make a mental note in case this ever becomes a bottleneck.